Add Nd support to clear_border

jmetz opened the pull request:
## Description
Adds 3d (in principal Nd) support to clear border. 


## References
#2086

ahojnnes commented:
Please, update the doc string to reflect the new N-D capability.

jmetz commented:
@ahojnnes - will do - the only reason I hadn't done this is that I saw that e.g. `regionprops` which accepts `3d` as of a while ago still has similar 2d-version of docstring so wasn't sure if that was left intentionally or not.

jmetz commented:
Umm is there a preference for the docstring? I am considering either e.g. :
```
labels : (N, M[, L,...]) array of int
        Label or binary image.
```
or 
```
labels : ND array of int
        Label or binary image.
```

soupault commented:
@jmetz as I'm aware, the preferable versions are `(M[, N[, ..., P]][, C])` for nD multichannel and `(M[, N[, ..., P]])` for nD single-channel.

jmetz commented:
Thanks @soupault ... changed! I realised the second line of the docstring appeared to 
have been left in by mistake and removed that too (as it's not an optional flag) - let me know if I should undo that.

ahojnnes commented:
LGTM, thanks!

soupault commented on skimage/segmentation/_clear_border.py:
```diff
@@ -5,12 +5,10 @@
-    labels : (N, M) array of int
-        Label or binary image.
+    labels : (M[, N[, ..., P]]) array of int
+        Label or binary image
```
This doesn't seem to be correct. I'd put something like:
```
labels : (M[, N[, ..., P]]) array of int or bool
    Imaging data labels.
```

jmetz replied:
@soupault - I didn't change the phrasing of that one.. can change it to `Imaging data labels` if that's preferred though.

soupault commented on skimage/segmentation/_clear_border.py:
```diff
@@ -21,7 +19,7 @@ def clear_border(labels, buffer_size=0, bgval=0, in_place=False):
     -------
-    labels : (N, M) array
+    labels : (M[, N[, ..., P]]) array
         Cleared binary image.
```
`Cleared binary image.`?

soupault replied:
Also, I'd change the name of output variable from `labels` to `out`.

jmetz replied:
As above - I simply maintained the previous docstrings in those respects, but if your suggested changes are preferred I can certainly include them in this pull request... or would it make more sense to do that in a different pull request?

soupault replied:
I'll be very grateful if you could change this. In the current redaction the descriptions are too confusing.
If not, I'll file another issue for this.

jmetz replied:
Sure no problem

soupault commented:
@jmetz perfect, thanks! Waiting for the test to pass.

jmetz commented:
Hey @soupault & @ahojnnes  - sorry to push for this but just so this doesn't end up being in conflict with the the repo... do you think it's good to merge?

soupault commented:
@jmetz thanks for the contribution!
